ksqldb-common/src/main/java/io/confluent/ksql/logging/processing/ProcessingLoggerWithMetricsInstantiator.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.logging.processing;
+
+import static java.util.Objects.requireNonNull;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import io.confluent.common.logging.StructuredLogger;
+import java.util.Map;
+import java.util.Objects;
+import org.apache.kafka.common.metrics.Metrics;
+
+public class ProcessingLoggerWithMetricsInstantiator {
```

Comment:
what's this class for? It seems to be just used in the implementation of `getLoggerWithMetrics`

Reply:
I used it as a wrapper class for everything needed for a `ProcessingLoggerImplWithMetrics`. It's so I can use a function type for `loggerWithMetricsFactory` in `ProcessingLoggerFactoryImpl`, otherwise I'd need need some special QuadFunction interface

Reply:
Ok I think, I can get away with a TriFunction after looking at some of your other suggestions, I can get rid of this class.